optimum/exporters/onnx/base.py

```diff
@@ -0,0 +1,437 @@
+
+from transformers.utils import is_tf_available, is_torch_available, is_vision_available
+
+from ...utils import logging
+from ...utils.input_generators import DummyInputGenerator
+from ..base import ExportConfig
+from .utils import MIN_TORCH_VERSION as GLOBAL_MIN_TORCH_VERSION
+from .utils import ParameterFormat, compute_serialized_parameters_size
+
+
+if TYPE_CHECKING:
+    from transformers import PretrainedConfig
+
+if is_torch_available():
+    import torch
+
+if is_tf_available():
+    import tensorflow as tf
+
+if is_vision_available():
+    from PIL import Image
+
+logger = logging.get_logger(__name__)
+
+
+# 2 Gb
+EXTERNAL_DATA_FORMAT_SIZE_LIMIT = 2 * 1024 * 1024 * 1024
+
+
+@dataclasses.dataclass
+class PatchingSpec:
+    """
+    Data class that holds patching specifications.
+
+    Args:
+        o: Module / object where the op to patch is located
+        name: Name of the op to monkey patch
+        custom_op: Custom op that patches the original op
+        orig_op: Original op that is being patched
+        op_wrapper: Wrapper (optional) that wraps both the original and custom ops.
+            It is useful for ops that are class or static methods for instance.
+    """
+
+    o: Any
+    name: str
+    custom_op: Callable
+    orig_op: Optional[Callable] = None
+    op_wrapper: Optional[Callable] = None
+
+
+class OnnxConfig(ExportConfig, ABC):
+    """
+    Base class for ONNX exportable model describing metadata on how to export the model through the ONNX format.
+
+    Class attributes:
+
+    - NORMALIZED_CONFIG_CLASS (`Type`) -- A class derived from [`~optimum.utils.NormalizedConfig`] specifying how to
+    normalize the model config.
+    - DUMMY_INPUT_GENERATOR_CLASSES (`Tuple[Type]`) -- A tuple of classes derived from
+    [`~optimum.utils.DummyInputGenerator`] specifying how to create dummy inputs.
+    - ATOL_FOR_VALIDATION (`Union[float, Dict[str, float]]`) -- A float or a dictionary mapping task names to float,
+    where the float values represent the absolute tolerance value to use during model conversion validation.
+    - DEFAULT_ONNX_OPSET (`int`, defaults to 11) -- The default ONNX opset to use for the ONNX export.
+    - MIN_TORCH_VERSION (`packaging.version.Version`, defaults to [`~optimum.exporters.onnx.utils.MIN_TORCH_VERSION`]) -- The
+    minimum torch version supporting the export of the model to ONNX.
+
+    Args:
+        config (`transformers.PretrainedConfig`):
+            The model configuration.
+        task (`str`, defaults to `"default"`):
+            The task the model should be exported for.
+    """
+
+    NORMALIZED_CONFIG_CLASS = None
+    DUMMY_INPUT_GENERATOR_CLASSES = ()
+    DEFAULT_ONNX_OPSET = 11
+    ATOL_FOR_VALIDATION: Union[float, Dict[str, float]] = 1e-5
+    MIN_TORCH_VERSION = GLOBAL_MIN_TORCH_VERSION
+    _TASK_TO_COMMON_OUTPUTS = {
+        "causal-lm": OrderedDict({"logits": {0: "batch", 1: "sequence"}}),
```

Comment:
Should we take the opportunity to name these axes the same way they are described in the `transformers` doc? 

E.g.

```suggestion
        "causal-lm": OrderedDict({"logits": {0: "batch_size", 1: "sequence_length"}}),
```

Personally, I find that being explicit really helps onboard new contributors who often ask me what should be put here for a new "feature"